README.md

```diff
@@ -1,16 +1,94 @@
 # cosi-driver-ceph
 
-Sample Driver that provides reference implementation for Container Object Storage Interface (COSI) API
+Sample Driver that provides reference implementation for Container Object Storage Interface (COSI) API for ceph object store aka Rados Gateway(RGW)
 
-## Community, discussion, contribution, and support
+#### Installing CRDS, COSI controller, Node adapter
+```
+$ kubectl create -k github.com/kubernetes-sigs/container-object-storage-interface-api
 
-Learn how to engage with the Kubernetes community on the [community page](http://kubernetes.io/community/).
+$ kubectl create -k github.com/kubernetes-sigs/container-object-storage-interface-controller
+
+$ kubectl create -k github.com/kubernetes-sigs/container-object-storage-interface-csi-adapter
+```
+Following pods will running on default namespace :
+```
+NAME                                        READY   STATUS    RESTARTS   AGE
+objectstorage-controller-6fc5f89444-4ws72   1/1     Running   0          2d6h
+objectstorage-csi-adapter-wsl4l             3/3     Running   0          2d6h
+```
+
+
+#### Building, Installing, Setting Up
+Code can be compiled using:
+```
+$ make build
+```
+Now build docker image and provide tag as `ceph/ceph-cosi-driver:latest`
+```
+$ docker build ./
+Sending build context to Docker daemon  41.95MB
+Step 1/5 : FROM gcr.io/distroless/static:latest
+ ---> 1d9948f921db
+Step 2/5 : LABEL maintainers="Kubernetes COSI Authors"
+ ---> Using cache
+ ---> 8659e9813ec5
+Step 3/5 : LABEL description="Ceph COSI driver"
+ ---> Using cache
+ ---> 0c55b21ff64f
+Step 4/5 : COPY ./bin/ceph-cosi-driver ceph-cosi-driver
+ ---> a21275402998
+Step 5/5 : ENTRYPOINT ["/ceph-cosi-driver"]
+ ---> Running in 620bfa992683
+Removing intermediate container 620bfa992683
+ ---> 09575229056e
+Successfully built 09575229056e (set this to "IMAGE_ID")
+
+docker tag $IMAGE_ID ceph/ceph-cosi-driver:latest
+```
+Now start the sidecar and cosi driver with:
+```
+$ kubectl apply -k .
+$ kubectl -n ceph-cosi-driver get pods
+NAME                                         READY   STATUS    RESTARTS   AGE
+objectstorage-provisioner-6c8df56cc6-lqr26   2/2     Running   0          26h
+```
+
+#### Create Bucket Requests, Bucket Access Request and consuming it on App
+```
+$ kubectl create -f examples/bucketclass.yaml
+$ kubectl create -f examples/bucketrequest.yaml
+$ kubectl create -f examples/bucketaccessclass.yaml
+$ kubectl create -f examples/bucketaccessrequest.yaml
+```
+In the app, `bucketaccessrequest(bar)` can be cosumed as volume mount:
```

Comment:
use `BucketAccessRequest (BAR)`

Reply:
typo: "consumed"

Reply:
Try this to replace above sentence:
"In the app pod, the `BucketAccessRequest (BAR) can be consumed as a volume mount:"